tensorflow/python/eager/def_function_test.py

```diff
@@ -239,6 +239,35 @@ def apply(self, x):
 
     m1 = MyModel()
     self.assertAllEqual(m1.apply(3.0), 6.0)
+  
+  def testMethodVariable(self):
+
+    class Foo:
+
+      def __init__(self):
+        self._flag_keyed_vars = {}
+        self.trace_count = 0
+
+      def __call__(self, var_creation_flag):
+        self.compute(var_creation_flag)
+        return self._flag_keyed_vars[var_creation_flag]
+
+      @def_function.function
+      def compute(self, var_creation_flag):
+        self.trace_count += 1
+        if var_creation_flag not in self._flag_keyed_vars:
+          if var_creation_flag:
+            self._flag_keyed_vars[var_creation_flag] = variables.Variable(1.0)
+          else:
+            self._flag_keyed_vars[var_creation_flag] = variables.Variable(2.0)
+    foo = Foo()
+    self.assertAllEqual(foo(True), 1.0)
+    self.assertEqual(foo.trace_count, 2)
+    self.assertAllEqual(foo(True), 1.0)
+    self.assertEqual(foo.trace_count, 2)
+    if def_function.PR_49310:
```

Comment:
Here, let's write two tests, one which overrides the flag to True, the other overriding it to False. That way the tests remain independent of any future changes, and we avoid the "if this test that", which is a bit of an antipattern.

Reply:
Now we have a parametrized test that it is failing in the FALSE case. Is this what you want or do you want that the test passes in both cases? In that case we need to totally separate this in two tests and in the false case expect `ValueError:` that is the current behavior in master with false.

Reply:
Yes, the test should expect a ValueError in the False case.

Reply:
OK, done